currency.sol

```diff
@@ -256,134 +281,176 @@ contract cccur is owned {
     
     // ensure the ETH level of the account
+    refill();
     topUp(_targetAccount);
-    topUp(msg.sender);
   }
+
+  function AllowReplaceBy(address target) public payable {
+     if (!actif) revert();                                                      // panic lock
+     if (newAddress[msg.sender]!=address(0)) revert();                          // Already replaced account cannot be replaced again
+     if (!accountStatus[msg.sender]) revert();                                  // locked account cannot be replaced
+     if (accountAlreadyUsed[target]==true) revert();                            // only new account can be a replacement target
+     if (ReplacementRequestNumber[msg.sender]>2) revert();                      // limit the number of replacement request possible
+
+     if (requestReplacementTo[target]!=address(0)) {                            // Cancel replacement request if exists
+       requestReplacementFrom[requestReplacementTo[target]]=address(0);
+       requestReplacementTo[target]=address(0);
+     }
+
+     if (requestReplacementFrom[msg.sender]!=address(0)) {                      // Cancel replacement request if exists
+       requestReplacementTo[requestReplacementFrom[msg.sender]]=address(0);
+       requestReplacementFrom[target]=address(0);
+     }
+
+     requestReplacementFrom[target]=msg.sender;                                // register the request   
+     requestReplacementTo[msg.sender]=target;
+     
+     ReplacementRequestNumber[msg.sender]+=1;
+     topUp(target);                                                            // ensure targuet has eth to accept the request
+  } 
+
+  function CancelReplaceBy() public  {
+     if (!actif) revert();                                                      // panic lock
+     if (requestReplacementFrom[msg.sender]!=address(0)) {                      // Cancel replacement request if exists
+       requestReplacementTo[requestReplacementFrom[msg.sender]]=address(0);
+       requestReplacementFrom[msg.sender]=address(0);
+       refill();  
+     }
+
+  }
+  
   
   /* replace the current account by a new one transfering its content */
   
-  function replaceAccount(address _replacementAccount) public {
+  function AcceptReplaceAccount() public {
      if (!actif) revert();                                                      // panic lock
-     if (newAddress[msg.sender]!=address(0)) revert();                          // Already replaced account cannot be replaced again
-     if (!accountStatus[msg.sender]) revert();                                  // locked account cannot be replaced
+     if (accountAlreadyUsed[msg.sender]==true) revert();                        // only new account can be a replacement target
+    
+     if (requestReplacementTo[msg.sender]==address(0)) revert();                // only existing request can be treated
```

Comment:
This is the check that doesn't seem consistent to me with what happens in `AllowReplaceBy()`.

Reply:
outdated